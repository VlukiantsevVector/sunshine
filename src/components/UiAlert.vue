<template>
  <div :class="classes">
    <div class="ui-alert__main">
      <slot />
    </div>

    <div class="ui-alert__buttons">
      <slot name="buttons" />
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import { classHelper } from '@/lib/utils';

const alertClassHelper = classHelper('ui-alert');


export default Vue.extend({
  name: 'ui-alert',

  props: {
    success: {
      type: Boolean,
      default: false,
    },
    warning: {
      type: Boolean,
      default: false,
    },
    error: {
      type: Boolean,
      default: false,
    },
  },

  computed: {
    classes(): object {
      return alertClassHelper({
        success: this.success,
        warning: this.warning,
        error: this.error,
      });
    },
  },
});
</script>

