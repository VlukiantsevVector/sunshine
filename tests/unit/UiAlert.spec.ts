import { mount } from '@vue/test-utils';
import UiAlert from '@/components/UiAlert.vue';

describe('UiAlert', () => {
  it('renders successful alert', () => {
    const wrapper = mount(UiAlert, {
      propsData: { success: true },
      slots: {
        default: 'Success message',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders warning alert', () => {
    const wrapper = mount(UiAlert, {
      propsData: { warning: true },
      slots: {
        default: 'Warning message',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders error alert', () => {
    const wrapper = mount(UiAlert, {
      propsData: { error: true },
      slots: {
        default: 'Error message',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders button', () => {
    const wrapper = mount(UiAlert, {
      propsData: { warning: true },
      stubs: ['ui-button'],
      slots: {
        default: 'Warning message',
        buttons: `
          <ui-button small>Foo</ui-button>
        `,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
