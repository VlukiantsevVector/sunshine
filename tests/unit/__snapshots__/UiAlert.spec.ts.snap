// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiAlert renders button 1`] = `
<div class="ui-alert ui-alert--warning">
  <div class="ui-alert__main">Warning message</div>
  <div class="ui-alert__buttons">
    <ui-button-stub small="">Foo</ui-button-stub>
  </div>
</div>
`;

exports[`UiAlert renders error alert 1`] = `
<div class="ui-alert ui-alert--error">
  <div class="ui-alert__main">Error message</div>
  <div class="ui-alert__buttons"></div>
</div>
`;

exports[`UiAlert renders successful alert 1`] = `
<div class="ui-alert ui-alert--success">
  <div class="ui-alert__main">Success message</div>
  <div class="ui-alert__buttons"></div>
</div>
`;

exports[`UiAlert renders warning alert 1`] = `
<div class="ui-alert ui-alert--warning">
  <div class="ui-alert__main">Warning message</div>
  <div class="ui-alert__buttons"></div>
</div>
`;
